src/posit/connect/permissions.py

```diff
@@ -0,0 +1,107 @@
+from __future__ import annotations
+
+from typing import List, overload
+
+from requests.sessions import Session as Session
+
+from . import urls
+
+from .config import Config
+from .resources import Resource, Resources
+
+
+class Permission(Resource):
+    @property
+    def guid(self) -> str:
+        """The globally unique identifier.
+
+        An alias for the field 'id'.
+
+        The conventional name for this field is 'guid'. As of v2024.03.0, the API uses the field name 'id'.
+
+        Returns
+        -------
+        str
+        """
+        # Alias 'id' to 'guid'.
+        # The conventional name for this field across applications is 'guid'.
+        return self.get("id", self.get("guid"))  # type: ignore
```

Comment:
*Is* the id a GUID? Doesn't look like it according to the API docs, but the example there could be wrong.

Should we file an issue on Connect to correct this inconsistency?

Reply:
Ah, good catch. I just assumed it was a GUID. 👍🏻 

It's probably a discussion for another time, but I'm surprised to see that it's sequential. 

side note; I noticed that the request body schema definition here is incorrect. The example is correct, but the schema definition is missing the required `principal` fields. I'll file a ticket for it once I wrap things up.

Reply:
It would help if I provided the link: https://docs.posit.co/connect/api/#put-/v1/content/-guid-/permissions/-id-